src/utils/csv.ts

```diff
@@ -84,3 +87,65 @@ export const formatVariables = (row: string[], keys: string[], startIndex: numbe
     type: Array.isArray(value) ? 'array' : typeof value,
   }));
 };
+
+/**
+ * Generates a cartesian product of arrays.
+ * @param arrays The arrays to generate the product from.
+ * @returns The generated product.
+ */
+export const cartesianProduct = (arrays: any[][]): any[][] => {
```

Comment:
Should be able to use a generic type "<T>" instead of "any".  `<T>(arrays: T[][]): T[][]`